pkg/deploy/elbv2/load_balancer_synthesizer.go

```diff
@@ -114,6 +114,7 @@ func (s *loadBalancerSynthesizer) findSDKLoadBalancers(ctx context.Context) ([]L
 	stackTags := s.trackingProvider.StackTags(s.stack)
 	stackTagsLegacy := s.trackingProvider.StackTagsLegacy(s.stack)
 	return s.taggingManager.ListLoadBalancers(ctx,
+		stackTags,
```

Comment:
i think we don't pass this additional stackTags.

why not directly convert TagFilter to RGTFilter? we can add a method to [`TagFilter `](https://github.com/kubernetes-sigs/aws-load-balancer-controller/blob/main/pkg/deploy/tracking/tag_filter.go#LL7C8-L7C8) to convertItToRGTFilter.

since this ListLoadBalancers's interface accepts a list of tag filters, we can do as following:
```
lbARNS = sets.NewString().          # need to use a set here to avoid conflicts since same arn can be returned multiple times.
for _, tagFilter in tagFilters {
   resources, err = GetResources(...tagFilter)
  if err != nil {
     return err;
   }
   for _, resource in resources {
     lbARNS.insert(resource.arn)
  }
}
for _, lbARN := lbARNS {
......original other logic...
}
```
This way we will still discover the resources tagged with "legacyTags"(v1 controller), so remains our backwards compability as well. (only downside is we will be doing two RGT calls but are fine to me, we can remove legacyTags from the input once we decide to drop this v1->v2 migration support)

Reply:
Reverted the addition of stackTags, thanks for the hint! However I just keeps the resources in the updated PR, because we still need the `resource.Tags` in addtion to the `resource.ResourceARN`